### PR TITLE
support customized labels

### PR DESCRIPTION
**Pull request by mingmcb**
support customized labels

Master Issue: #156

### Motivation

to add allow user add customized labels

### Modifications

Update the deployment template for new labels

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
- [x] `no-need-doc` 
  this change is self-explain
- [ ] `doc`

### CODE REVIEW

**Comment by ericsyh** [comment]
@mingmcb thx for your contribution! just few suggestions for this PR:
1. Can we rename to use `labels` and `podLabels`? which I think will be more consistent with some existing fields like the `podAnnotations`. 
2. We should also add these two fields in the https://github.com/streamnative/pulsar-resources-operator/blob/main/charts/pulsar-resources-operator/values.yaml then other users can notice them.

**Review by ericsyh — APPROVED** [review]
LGTM

File: charts/pulsar-resources-operator/templates/deployment.yaml [hunk 1]
```diff
@@ -20,6 +20,7 @@ metadata:
   namespace: {{ include "pulsar-resources-operator.namespace" .}}
   labels:
     {{- include "pulsar-resources-operator.labels" . | nindent 4 }}
+    {{- toYaml .Values.labels" | nindent 4 }}
```
Comment (ericsyh) [inline charts/pulsar-resources-operator/templates/deployment.yaml:23]:
```suggestion
    {{- toYaml .Values.labels | nindent 4 }}
```

File: charts/pulsar-resources-operator/templates/deployment.yaml [hunk 2]
```diff
@@ -31,6 +32,7 @@ spec:
         {{- toYaml .Values.podAnnotations | nindent 8 }}
       labels:
         {{- include "pulsar-resources-operator.selectorLabels" . | nindent 8 }}
+        {{- toYaml .Values.podLabels" | nindent 8 }}
```
Comment (ericsyh) [inline charts/pulsar-resources-operator/templates/deployment.yaml:35]:
```suggestion
        {{- toYaml .Values.podLabels | nindent 8 }}
```

**Review by ericsyh — COMMENTED** [review]
Should remove unexpected `"` in the templates.

File: charts/pulsar-resources-operator/templates/deployment.yaml (hunk 1 above)
Comment (mingmcb) [reply charts/pulsar-resources-operator/templates/deployment.yaml]: good catch! please review

File: charts/pulsar-resources-operator/Chart.yaml [hunk 3]
```diff
@@ -30,13 +30,13 @@ type: application
...
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: v0.4.3
+version: v0.4.4
 
 # This is the version number of the application being deployed. This version number should be
 # incremented each time you make changes to the application. Versions are not expected to
 # follow Semantic Versioning. They should reflect the version the application is using.
 # It is recommended to use it with quotes.
-appVersion: "v0.4.3"
+appVersion: "v0.4.4"
```
Comment (ericsyh) [inline charts/pulsar-resources-operator/Chart.yaml:39]: @mingmcb `appVersion` means the operator image version here. 
So for your case,  should not change the `appVersion` here cause your change is only about the chart template and won't need a new operator image.

**Review by ericsyh — COMMENTED** [review]
left a comment.

**Review by ericsyh — APPROVED** [review]
LGTM

**Pull request merged**